### PR TITLE
ZER-52 - Add Capybara test for Password Reset page

### PR DESCRIPTION
**Pull request by halyna-m**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [x] @github_username

### Second Level Review

- [x] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [x]  СI passed
- [x]  Сode coverage >=95%
- [x]  PR is reviewed manually again (to make sure you have 100% ready code)
- [x]  All reviewers agreed to merge the PR
- [x]  I've checked new feature as logged in and logged out user if needed
- [x]  PR meets all conventions

### CODE REVIEW

**Review by olenochka123 — COMMENTED** [review]
everything's good)

**Review by andriibureviy — APPROVED** [review]
Everything is fine

**Review by olenochka123 — APPROVED** [review]

File: spec/features/reset_password_spec.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      click_link 'Log in'
+      expect(page).to have_selector("a[href='/users/sign_in']")
+    end
+    it "redirect to sign in page after click the button sign up" do
+      visit '/users/password/new' 
+      click_link 'Sign up'
+      expect(page).to have_selector("a[href='/users/sign_up']")
+    end
+  end
+end
```
Comment (maksymst) [inline spec/features/reset_password_spec.rb:28]: Add new line at EOF

File: spec/features/reset_password_spec.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    it "redirect to sign in page after click the button log in" do
+      visit '/users/password/new' 
+      click_link 'Log in'
+      expect(page).to have_selector("a[href='/users/sign_in']")
+    end
+    it "redirect to sign in page after click the button sign up" do
+      visit '/users/password/new' 
+      click_link 'Sign up'
+      expect(page).to have_selector("a[href='/users/sign_up']")
+    end
```
Comment (maksymst) [inline spec/features/reset_password_spec.rb:26]: 1. the text in `it` is confusing
2. user visits password reset page, clicks link `Sign Up` and sees another `Sign Up` link. This test is not really verifying anything. Let's try to improve this scenario

File: spec/features/reset_password_spec.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+      allow(Devise::Mailer).to receive(:reset_password_instructions).and_return(double(deliver: true))
+      visit '/users/password/new' 
+      fill_in 'Email', with: user.email
+      click_button 'Send me reset password instructions'
+      expect(page).to have_selector("a[href='/users/sign_in']")
+    end
+    it "redirect to sign in page after click the button log in" do
+      visit '/users/password/new' 
+      click_link 'Log in'
+      expect(page).to have_selector("a[href='/users/sign_in']")
```
Comment (maksymst) [inline spec/features/reset_password_spec.rb:20]: See coment below

File: spec/features/reset_password_spec.rb [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  let(:user) { create(:user) }
+  let(:calculator) { create(:calculator) }
+  context "when reset password" do
+    it "redirect to sign in page after click the button send me instruction" do
+      allow_any_instance_of(ApplicationController).to receive(:after_sign_in_path_for).and_return("/calculators/#{calculator.slug}")
+      allow(Devise::Mailer).to receive(:reset_password_instructions).and_return(double(deliver: true))
+      visit '/users/password/new' 
+      fill_in 'Email', with: user.email
+      click_button 'Send me reset password instructions'
+      expect(page).to have_selector("a[href='/users/sign_in']")
```
Comment (maksymst) [inline spec/features/reset_password_spec.rb:15]: See comment below

**Review by maksymst — CHANGES_REQUESTED** [review]
There is an issue with test describing. 
Rspec has several methods that can build a kind of documentation based on tests.

```ruby
describe '#some_method'
# OR
describe SomeClass
# OR
describe 'Some Page'
```
`context` and `it` together build the sentence
```ruby
context 'when user clicks button' do
  it 'redirects to home page' { ... }
end

# OR

context `when parameter is passed` do
  context 'when parameter is string' do
    It 'returns nil' { ... }
  end
end
```

Try to fix your tests according to these rules

**Review by andriibureviy — APPROVED** [review]

**Review by olenochka123 — APPROVED** [review]

**Review by andriibureviy — APPROVED** [review]

**Review by andriibureviy — APPROVED** [review]

**Review by olenochka123 — APPROVED** [review]

File: spec/features/reset_password_spec.rb [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+        .and_return(double(deliver: true))
+      visit PASSWORD_RESET_PATH
+      fill_in 'Email', with: user.email
+      click_button 'Send me reset password instructions'
+      expect(page).to have_content('If your email address exists')
+    end
+  end
+end
+
+describe 'Log In', js: true do
```
Comment (maksymst) [inline spec/features/reset_password_spec.rb:24]: Leave the test but remove `describe`

File: spec/features/reset_password_spec.rb [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+  context 'when user clicks button Log in' do
+    it 'redirect to sign in page' do
+      visit PASSWORD_RESET_PATH
+      click_link 'Log in'
+      expect(page).to have_current_path('/users/sign_in')
+    end
+  end
+end
+
+describe 'Sign Up', js: true do
```
Comment (maksymst) [inline spec/features/reset_password_spec.rb:34]: Leave the test but remove `describe`

File: spec/features/reset_password_spec.rb [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+  context 'when user clicks button Sign up' do
```
Comment (maksymst) [inline spec/features/reset_password_spec.rb:35]:
```suggestion
  context 'when user clicks Sign up link' do
```

File: spec/features/reset_password_spec.rb [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+      visit PASSWORD_RESET_PATH
+      fill_in 'Email', with: user.email
+      click_button 'Send me reset password instructions'
+      expect(page).to have_content('If your email address exists')
+    end
+  end
+end
+
+describe 'Log In', js: true do
+  context 'when user clicks button Log in' do
```
Comment (maksymst) [inline spec/features/reset_password_spec.rb:25]:
```suggestion
  context 'when user clicks Log in link' do
```

**Review by maksymst — CHANGES_REQUESTED** [review]

**Review by andriibureviy — APPROVED** [review]

**Review by olenochka123 — APPROVED** [review]

**Review by maksymst — APPROVED** [review]

**Pull request merged**